src/legacy/core_plugins/expressions/public/np_ready/public/loader.ts

```diff
@@ -128,18 +130,22 @@ export class ExpressionLoader {
     expression: string | ExpressionAST,
     params: IExpressionLoaderParams
   ): Promise<void> => {
-    this.loadingSubject.next();
-    if (this.dataHandler) {
+    if (this.dataHandler && this.dataHandler.isPending()) {
+      this.ignoreNextResponse = true;
       this.dataHandler.cancel();
     }
     this.setParams(params);
     this.dataHandler = new ExpressionDataHandler(expression, params);
+    if (!params.inspectorAdapters) params.inspectorAdapters = this.dataHandler.inspect();
     const data = await this.dataHandler.getData();
+    if (this.ignoreNextResponse) {
```

Comment:
we only want to swallow the error in case request was aborted internally because we are making a new one.

Reply:
This is the only thing I couldn't figure out how to test.
Would appreciate a hint.

Reply:
testing swallowing errors:
- go to dashboard or visualize
- open dev tools go to network panel
- click refresh few times (you should see requests being aborted, but chart should not show any errors)

testing not swallowing errors:
- there is no way to test request aborting and showing errors atm. this will only happen once we have cancel button in dashboard.
- you can test that any other errors are not being swallowed. maybe killing elasticsearch instance and then clicking refresh button.

Reply:
I used network throttling to slow things down, but requests are still not being canceled in network panel.

![image](https://user-images.githubusercontent.com/3016806/68601452-03edf080-04ad-11ea-9009-5c815820b9d7.png)